conans/server/revision_list.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+import time
+
+from conans.errors import ConanException
+
+
+class RevisionList(object):
```

Comment:
This object is a bit weird. Why not extending a ``list`` directly?

Reply:
Ok, but I will serialize/deserialize the object as a/from a dictionary with a key because:

1. We always regret not being able to add new fields.
2. Not using a JSON in a serialization is an error we made many times
3. As a general rule, a JSON is better (secure) as a dict instead of a list.

These were the reasons I modeled the object like that.

Reply:
ah! 4th reason. I hate to subclass a native type and add 200 magic methods to override an "add" or a "remove". It was never practical.

Reply:
I agree with the Json serialization, I was talking about the in-memory data access model. Agree that composition would be preferred over inheritance